src/core/editor/pdf_editor.js

```diff
@@ -431,6 +707,293 @@ class PDFEditor {
     }
   }
 
+  #visitObject(obj, callback, visited = new RefSet()) {
+    if (obj instanceof Ref) {
+      if (!visited.has(obj)) {
+        visited.put(obj);
+        this.#visitObject(this.xref[obj.num], callback, visited);
+      }
+      return;
+    }
+    if (Array.isArray(obj)) {
+      for (const item of obj) {
+        this.#visitObject(item, callback, visited);
+      }
+      return;
+    }
+    let dict;
+    if (obj instanceof BaseStream) {
+      ({ dict } = obj);
+    } else if (obj instanceof Dict) {
+      dict = obj;
+    }
+    if (dict) {
+      callback(dict);
+      for (const [, value] of dict.getRawEntries()) {
```

Comment:
If we're only interested in the values we can probably use `dict.getRawValues()` here instead to simplify the code a bit.